enhancements/annotation-storage.md

```diff
@@ -0,0 +1,126 @@
+* Annotations that have special keys such as `quay.expires-after=2d` should have the same behavior
+as labels.
+* UI should differentiate between labels and annotations
+
+### Non-Goals
+* Users can remove annotations from manifests
+* Users can see annotations when viewing labels
+
+## Proposal
+
+Make use of the jsonb datatype in postgres to store manifests. This will make it easier
+to parse the manifest for the annotations so they are more readily available. The `manifestbytes`
+column should be changed to a json type since this is the column where annotations are to be found.
+
+There would need to be another method added to the database model to obtain the annotations from this
+column once it is converted to a json type. The current way of listing labels can also be updated to 
+refer to the manifest json as well.
+
+### Risks and Mitigations
+
+A database migration is required for this change. Since it will be changing a column, there needs to be
+checks to make sure that it is safe for the user to apply this upgrade. 
+
+## Design Details
+
+Currently the manifest_bytes column in the manifest table is of type text
+
+*Manifest Table*
+
+|         Column         |          Type          | Collation | Nullable |               Default                | Storage  | Stats target | Description
+|------------------------|------------------------|-----------|----------|--------------------------------------|----------|--------------|-------------
+| id                     | integer                |           | not null | nextval('manifest_id_seq'::regclass) | plain    |              |
+| repository_id          | integer                |           | not null |                                      | plain    |              |
+| digest                 | character varying(255) |           | not null |                                      | extended |              |
+| media_type_id          | integer                |           | not null |                                      | plain    |              |
+| manifest_bytes         | text                   |           | not null |                                      | extended |              |
+| config_media_type      | character varying(255) |           |          |                                      | extended |              |
+| layers_compressed_size | bigint                 |           |          |                                      | plain    |              |
+
+The type can be changed to JSONB using an alembic revision. JSONB should be used instead
+of JSON because JSONB is significantly faster to process. This will reduce overhead when 
+retrieving manifest annotations.
+
+A new endpoint will need to be added that allows users to access manifest annotations.
+It should be similar to the labels endpoint to avoid confusion.
+
+**Get annotation(s)**
+```
+GET /api/v1/repository/{repository}/manifest/{manifestref}/annotations
+```
+
+**Update annotation**
```

Comment:
annotations can be in multiple places in the manifest. The most obvious one is  at the top layer `.annotations` but there also be annotations on each layer and also in the `config` we should provide the ability to query/update them as well

Reply:
Ok will those also be part of the `manifest_bytes` column?